src/main/template/vpc.yaml

```diff
@@ -0,0 +1,410 @@
+AWSTemplateFormatVersion: 2010-09-09
+Description: This template deploys a VPC. Its IP range is quartered into four parts. The first three parts form private
+  subnets, the fourth is again quartered into three public subnets. Both the private and the public subnets are then
+  spread across three Availability Zones. All the required infrastructure, like internet gateway, nat gateways, and
+  default routes are included. Also, a bastion jump host is deployed in order to make any private machines reachable.
+
+Parameters:
+  cidrBlock:
+    Description: The CIDR range of the VPC. Note that it cannot be larger than 16 bits.
+    Type: String
+    Default: 10.0.0.0/16
+
+  firstQuarteringPartitionBits:
+    Description: The bit size of the first quartering, which is the size of the three private subnets. This is the size
+      of the CIDR block minus two. Unfortunately, CloudFormation does not support arithmetics, so this needs to be
+      passed in as a parameter.
+    Type: String
+    Default: 14
+
+  secondQuarteringPartitionBits:
+    Description: The bit size of the second quartering, which is the size of the three public subnets. This is the size
+      of the CIDR block minus four. Unfortunately, CloudFormation does not support arithmetics, so this needs to be
+      passed in as a parameter.
+    Type: String
+    Default: 12
+
+  bastionAmi:
+    Description: The AMI used to create the bastion jump host. By default, Amazon Linux 2 is used.
+    Type: AWS::EC2::Image::Id
+    Default: ami-0f3a43fbf2d3899f7
+
+  keyName:
```

Comment:
Yes, we should. Placing this key on the instance will always work. Your custom setup which adds more keys might not. If it didn't, you can still diagnose the machine by using the account key. Otherwise, you have no way of telling what went wrong.